internal/controller/nodedisruption_controller.go

```diff
@@ -281,6 +281,19 @@ func (ndr *SingleNodeDisruptionReconciler) generateRejectedStatus(reason string)
 	}
 }
 
+// appendRejectedStatus add a new status on top of the existing ones
+func (ndr *SingleNodeDisruptionReconciler) appendRejectedStatus(reason string) []nodedisruptionv1alpha1.DisruptedBudgetStatus {
+	return append(ndr.NodeDisruption.Status.DisruptedDisruptionBudgets, nodedisruptionv1alpha1.DisruptedBudgetStatus{
+		Reference: nodedisruptionv1alpha1.NamespacedName{
+			Namespace: ndr.NodeDisruption.Namespace,
+			Name:      ndr.NodeDisruption.Name,
+			Kind:      ndr.NodeDisruption.Kind,
+		},
+		Reason: reason,
+		Ok:     false,
+	})
+}
```

Comment:
GenerateRejectedStatus create a brand new status list, this new function only append to a pre-existing lists

Reply:
I meant, there is duplication of the creation of the very same `nodedisruptionv1alpha1.DisruptedBudgetStatus` object in both case. The only diff is that `generateRejectedStatus` returns this new object as the only element of the list while the new function appends it to the existing disrupted budgets status. We can surely improve this.

Reply:
I made a commit that removes both

Reply:
Made a third iteration:
- `ValidateWithInternalConstraints` decides what to do with the status reported by each constraints
- Constraints within the internal constraints should return a single status
- To avoid de-duplication, we only add a method to fetch the reference. This avoids having a very specific function while retaining readability.

Reply:
@komuta any update?

Reply:
Sorry, LGTM